lms/djangoapps/courseware/tests/test_video_mongo.py

```diff
@@ -1824,3 +1837,144 @@ def test_bumper_metadata(self, get_url_for_profiles, get_bumper_settings, is_bum
+            expected_content = self.item_descriptor.xmodule_runtime.render_template('video.html', expected_context)
+
+        self.assertEqual(content, expected_content)
+
+    def change_course_setting_autoadvance(self, new_value):
+        """
+        Change the .video_auto_advance course setting (a.k.a. advanced setting).
+        This avoids doing .save(), and instead modifies the instance directly.
+        Based on test code for video_bumper setting.
+        """
+        # This first render is done to initialize the instance
+        self.item_descriptor.render(STUDENT_VIEW)
+        item_instance = self.item_descriptor.xmodule_runtime.xmodule_instance
+        item_instance.video_auto_advance = new_value
+        # After this step, render() should see the new value
+        # e.g. use self.item_descriptor.render(STUDENT_VIEW).content
+
+    def test_is_autoadvance_available_and_enabled(self):
+        """
+        Check that the autoadvance is not available when it is disabled via feature flag
+        (ENABLE_AUTOADVANCE_VIDEOS set to False) or by the course setting.
+        It checks that:
+        - only when the feature flag and the course setting are True (at the same time)
+          the controls are visible
+        - in that case (when the controls are visible) the video will autoadvance
+          (because that's the default), in other cases it won't
+        """
+        # Feature flag false and course setting false: not available (doubly disabled)
+        self.FEATURES.update({"ENABLE_AUTOADVANCE_VIDEOS": False})
+        self.change_course_setting_autoadvance(new_value=False)
+        self.assert_content_matches_expectations(
+            autoadvanceenabled_must_be=False,
+            autoadvance_must_be=False,  # because disabled
+        )
+
+        # Feature flag false and course setting true: not available (globally disabled)
+        self.FEATURES.update({"ENABLE_AUTOADVANCE_VIDEOS": False})
+        self.change_course_setting_autoadvance(new_value=True)
+        self.assert_content_matches_expectations(
+            autoadvanceenabled_must_be=False,
+            autoadvance_must_be=False,  # because disabled
+        )
+
+        # Enable flag but keep course setting disabled: not available (locally disabled)
+        self.FEATURES.update({"ENABLE_AUTOADVANCE_VIDEOS": True})
+        self.change_course_setting_autoadvance(new_value=False)
+        self.assert_content_matches_expectations(
+            autoadvanceenabled_must_be=False,
+            autoadvance_must_be=False,  # because disabled
+        )
+
+        # Enable flag, and course setting enabled: only case when it's available
+        self.FEATURES.update({"ENABLE_AUTOADVANCE_VIDEOS": True})
+        self.change_course_setting_autoadvance(new_value=True)
+        self.assert_content_matches_expectations(
+            autoadvanceenabled_must_be=True,  # controls visible
+            autoadvance_must_be=True,  # autoadvances by default
+        )
```

Comment:
This test could get more DRY with DDT, e.g.:

```python
    @ddt.data(
        (False, False),
        (False, True),
        (True, False),
        (True, True),
    )
    @ddt.unpack
    def test_is_autoadvance_available_and_enabled(self, global_setting, course_setting):
        """
        Check that the autoadvance is not available when it is disabled via feature flag
        (ENABLE_AUTOADVANCE_VIDEOS set to False) or by the course setting.
        It checks that:
        - only when the feature flag and the course setting are True (at the same time)
          the controls are visible
        - in that case (when the controls are visible) the video will autoadvance
          (because that's the default), in other cases it won't
        """
        # Feature flag false and course setting false: not available (doubly disabled)
        self.FEATURES.update({"ENABLE_AUTOADVANCE_VIDEOS": global_setting})
        self.change_course_setting_autoadvance(new_value=course_setting)
        self.assert_content_matches_expectations(
            autoadvanceenabled_must_be=(global_setting and course_setting),
            autoadvance_must_be=(global_setting and course_setting),  # because disabled
        )
```

Note to use DDT the test class also needs the `@ddt.ddt` decorator.

Reply:
Ok, I changed it, though I prefer readable tests even if they're longer: there's more place for comments, less logic, and failures are easier to understand. Uploaded.

Reply:
There's a tradeoff between being verbose and informative, but in this case it's enough code where it seems like a net gain to lean on the informative side. This test is extremely uncomplicated and basic as it's just verifying an "and" table: Only if the global and course settings are true, the auto-advance component flags are enabled.

With the expanded version, I had to bat my eyes around to verify it was testing all of the permutations, and additionally verify that something more dense wasn't being tested by ensuring all of the copied code was otherwise exactly the same. I feel if I come across this test in Open edX at some time in the future (for example, if I break the test in a PR), it will be easier for me to verify that the test is doing no more than what it claims.